File: webhole/packages/python-django/ls.py
def ls(path="."):
    import os
    import json
    try:
        path = os.path.normpath(os.path.abspath(path))

        # Ensure drive letters like "D:" are formatted as "D:\"
        if len(path) == 3 and path[1] == ":":
            path += r"\\"

        #path="D:\\"

        if not os.path.exists(path):
            return json.dumps([f"❌ Error: '{path}' not found."], ensure_ascii=False)

        items = os.listdir(path)
        result = [
            f"📁 {item}" if os.path.isdir(os.path.join(path, item)) else f"📄 {item}"
            for item in items
        ]
        return json.dumps(result, ensure_ascii=False)
    except Exception as e:
        return json.dumps([f"❌ Error: {str(e)}"], ensure_ascii=False)
    
print(ls(r'__path__'))

File: webhole/packages/go-http-server/ls.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"path/filepath"
)

func main() {
	path := "."
	if len(os.Args) > 1 {
		path = os.Args[1]
	}

	output := ls(path)
	fmt.Println(output)
}

func ls(path string) string {
	// Normalize the path
	absPath, err := filepath.Abs(path)
	if err != nil {
		errMsg, _ := json.Marshal([]string{fmt.Sprintf("❌ Error: %v", err)})
		return string(errMsg)
	}

	// Ensure drive like "D:" becomes "D:\"
	if len(absPath) == 2 && absPath[1] == ':' {
		absPath += `\`
	}

	// Check existence
	if _, err := os.Stat(absPath); os.IsNotExist(err) {
		errMsg, _ := json.Marshal([]string{fmt.Sprintf("❌ Error: '%s' not found.", absPath)})
		return string(errMsg)
	}

	// Read directory
	entries, err := os.ReadDir(absPath)
	if err != nil {
		errMsg, _ := json.Marshal([]string{fmt.Sprintf("❌ Error: %v", err)})
		return string(errMsg)
	}

	var result []string
	for _, entry := range entries {
		name := entry.Name()
		if entry.IsDir() {
			result = append(result, "📁 "+name)
		} else {
			result = append(result, "📄 "+name)
		}
	}

	jsonResult, err := json.Marshal(result)
	if err != nil {
		errMsg, _ := json.Marshal([]string{fmt.Sprintf("❌ Error: %v", err)})
		return string(errMsg)
	}

	return string(jsonResult)
}
